File: taurus-protect-sdk-python/taurus_protect/models/earn.py
"""Earn models for Taurus-PROTECT SDK."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class EarnReward(BaseModel):
    """
    A reward earned by an address.

    Attributes:
        id: Unique reward identifier.
        recipient_address_id: The internal address ID receiving the reward.
        recipient_address: The receiving blockchain address.
        reward_type: Reward type (e.g. ``RewardTypeMerklToken``).
        amount: Total reward amount.
        claimed: Amount already claimed.
        pending: Amount still pending.
        token_address: Contract address of the reward token.
        token_symbol: Symbol of the reward token.
        token_asset_id: Asset ID of the reward token.
    """

    id: str = Field(description="Unique reward identifier")
    recipient_address_id: Optional[str] = Field(default=None, description="Recipient address ID")
    recipient_address: Optional[str] = Field(default=None, description="Recipient address")
    reward_type: Optional[str] = Field(default=None, description="Reward type")
    amount: Optional[str] = Field(default=None, description="Total reward amount")
    claimed: Optional[str] = Field(default=None, description="Claimed amount")
    pending: Optional[str] = Field(default=None, description="Pending amount")
    token_address: Optional[str] = Field(default=None, description="Reward token address")
    token_symbol: Optional[str] = Field(default=None, description="Reward token symbol")
    token_asset_id: Optional[str] = Field(default=None, description="Reward token asset ID")

    model_config = {"frozen": True}

File: taurus-protect-sdk-typescript/src/models/earn.ts
/**
 * Earn models for Taurus-PROTECT SDK.
 *
 * Rewards earned by addresses, such as Merkl token incentives.
 */

import type { CursorPage, CursorPageOptions } from './pagination';

/** The token a reward is paid in. */
export interface EarnRewardToken {
  /** Token contract address */
  readonly address?: string;
  /** Token symbol */
  readonly symbol?: string;
  /** Asset ID of the token */
  readonly assetId?: string;
}

/** A Merkl token reward. */
export interface MerklTokenReward {
  /** Total reward amount */
  readonly amount?: string;
  /** Amount already claimed */
  readonly claimed?: string;
  /** Amount pending */
  readonly pending?: string;
  /** The token the reward is paid in */
  readonly token?: EarnRewardToken;
}

/**
 * A reward earned by an address.
 */
export interface EarnReward {
  /** Reward ID */
  readonly id?: string;
  /** Internal ID of the address receiving the reward */
  readonly recipientAddressId?: string;
  /** Blockchain address receiving the reward */
  readonly recipientAddress?: string;
  /** Reward type (e.g., "RewardTypeMerklToken") */
  readonly rewardType?: string;
  /** Details of a Merkl token reward */
  readonly merklTokenReward?: MerklTokenReward;
}

/**
 * Options for listing rewards. A cursor list: `pageSize` 1-100 (default 20) and the
 * `cursor` of a previous page.
 */
export interface ListEarnRewardsOptions extends CursorPageOptions {
  /** Keep the rewards of this recipient address */
  readonly recipientAddressId?: string;
}

/**
 * A page of rewards.
 */
export interface ListEarnRewardsResult {
  /** The rewards of this page */
  readonly items: EarnReward[];
  /** Cursor pagination: continue with `cursor: pagination.nextCursor` while `hasMore` */
  readonly pagination: CursorPage;
}
